### PR TITLE
decompile code_0804B9F8.s

### PR DESCRIPTION
**Pull request by destoer**
decompiled asm/code_0804B9F8.s

### CODE REVIEW

File: src/code_0804B9F8.c [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#include "global.h"
+#include "utils.h"
+#include "fileScreen.h"
+
+
+
+#define gPaletteRam ((u8*)(0x5000000))
```
Comment (notyourav) [inline src/code_0804B9F8.c:7]: This should be defined as a macro somewhere in the gba/ headers.

File: src/code_0804B9F8.c [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+extern u16 gPaletteBuffer[];
+extern u32 gUsedPalettes;
+
+// function pointer to overlay (0x03005e98) in ram calls rom function MakeFadeBuff256
+extern u32 gMakeFadeBuff256;
+typedef void (*fptrMakeFadeBuff256)(u8*,u8*,u16,u8);
+
+
+void FadeMain()
+{
```
Comment (notyourav) [inline src/code_0804B9F8.c:18]: Clang format please

**Review by notyourav — APPROVED** [review]
Lgtm just a few things

File: src/code_0804B9F8.c (hunk 2 above)
Comment (destoer) [reply src/code_0804B9F8.c]: sorry how do i make sure this gets run on every file because i screwed it up the first time whoops.

Comment (octorock) [reply src/code_0804B9F8.c]: If you have clang-format installed, you should be able to run the `format.sh` in the root folder of this repo.

**Pull request merged**